inception-ui-kb/src/main/java/de/tudarmstadt/ukp/inception/ui/kb/ConceptTreePanel.java

```diff
@@ -127,13 +127,34 @@ protected boolean isSelected()
         addLink.add(new Label("label", new ResourceModel("concept.list.add")));
         addLink.add(new WriteProtectionBehavior(kbModel));
         add(addLink);
+        /**
+         * Bind the button control to the click even and respond to the button's click event
+         */
+        add(new LambdaAjaxLink("autoCreateLabel", this::autoCreateLabel));
 
         Form<Preferences> form = new Form<>("form", CompoundPropertyModel.of(preferences));
         form.add(new CheckBox("showAllConcepts").add(
                 new LambdaAjaxFormSubmittingBehavior("change", this::actionPreferenceChanged)));
         add(form);
     }
-    
+
+    /**
+     * handle click event
+     */
+    protected void autoCreateLabel(AjaxRequestTarget aTarget) throws Exception {
+        List<KBHandle> list = kbService.listAllConcepts(kbModel.getObject(), true);
+        //go through each resource in the KB
+        for(KBHandle kbh : list) {
```

Comment:
This change is not related to this refactoring. Please create a new pull request for feature changes.